### ISSUE TITLE
REINFORCE++ global normalization uses a different standard deviation on one GPU than on many

### ISSUE THREAD

**Issue by CodersAcademy006** [ellanorai/ThinkRL#112]
`REINFORCEPPAlgorithm._global_statistics` at `thinkrl/algorithms/reinforce_pp.py:158` computes the standard deviation two different ways depending on whether the process group is initialized.

Single process, `:183`:

```python
if not dist.is_initialized():
    return tensor.mean(), tensor.std()
```

`torch.Tensor.std()` defaults to `correction=1`, the Bessel-corrected sample standard deviation, dividing by `n-1`.

Distributed, `:196-198`:

```python
global_var = (local_sq_sum / local_count) - (global_mean**2)
global_std = torch.sqrt(torch.clamp(global_var, min=1e-8))
```

This is the population standard deviation, dividing by `n`.

The same batch of advantages therefore gets normalized by a different divisor depending on world size, so a single-GPU run and a multi-GPU run are not comparable even with identical data and seeds. The gap is `sqrt(n/(n-1))`, which is under a percent for large batches but reaches 6% at `n=8` and 12% at `n=4`, and per-device batches in that range are normal for long-sequence reasoning work.

Global advantage normalization is one of the two things the module docstring names as REINFORCE++ features, so it is worth having it mean the same thing in both modes.

**Suggested fix**

Use `tensor.std(correction=0)` in the single-process branch, or apply the `n/(n-1)` correction in the distributed branch. Either is fine as long as the two agree; population std is the simpler choice since the distributed path would otherwise need an extra term.

I am happy to open a PR for this.
